diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/embeddedRDF.py pyRdfa/embeddedRDF.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/embeddedRDF.py	2012-03-23 14:32:43.000000000 +0000
+++ pyRdfa/embeddedRDF.py	2012-04-03 00:26:42.781131661 +0100
@@ -7,7 +7,7 @@
 @license: This software is available for use under the
 U{W3C® SOFTWARE NOTICE AND LICENSE<href="http://www.w3.org/Consortium/Legal/2002/copyright-software-20021231">}
 @contact: Ivan Herman, ivan@w3.org
-@version: $Id: embeddedRDF.py,v 1.9 2012/03/23 14:06:25 ivan Exp $
+@version: $Id: embeddedRDF.py,v 1.9 2012-03-23 14:06:25 ivan Exp $
 """
 
 from StringIO	 import StringIO
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/graph.py pyRdfa/graph.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/graph.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/graph.py	2012-04-03 00:26:42.777131637 +0100
@@ -14,7 +14,7 @@
 """
 
 """
-$Id: graph.py,v 1.6 2012/03/23 14:06:25 ivan Exp $ $Date: 2012/03/23 14:06:25 $
+$Id: graph.py,v 1.6 2012-03-23 14:06:25 ivan Exp $ $Date: 2012-03-23 14:06:25 $
 
 """
 
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/host/atom.py pyRdfa/host/atom.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/host/atom.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/host/atom.py	2012-04-03 00:26:42.781131661 +0100
@@ -12,8 +12,8 @@
 """
 
 """
-$Id: atom.py,v 1.1 2011/08/12 10:05:55 ivan Exp $
-$Date: 2011/08/12 10:05:55 $
+$Id: atom.py,v 1.1 2011-08-12 10:05:55 ivan Exp $
+$Date: 2011-08-12 10:05:55 $
 """
 
 def atom_add_entry_type(node, state) :
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/host/html5.py pyRdfa/host/html5.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/host/html5.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/host/html5.py	2012-04-03 00:26:42.781131661 +0100
@@ -12,8 +12,8 @@
 """
 
 """
-$Id: html5.py,v 1.7 2012/03/23 14:06:31 ivan Exp $
-$Date: 2012/03/23 14:06:31 $
+$Id: html5.py,v 1.7 2012-03-23 14:06:31 ivan Exp $
+$Date: 2012-03-23 14:06:31 $
 """
 
 # The handling of datatime is a little bit more complex... better put this in a separate function for a better management
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/host/__init__.py pyRdfa/host/__init__.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/host/__init__.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/host/__init__.py	2012-04-03 02:47:35.995048882 +0100
@@ -26,8 +26,8 @@
 """
 
 """
-$Id: __init__.py,v 1.11 2012/03/23 14:06:31 ivan Exp $
-$Date: 2012/03/23 14:06:31 $
+$Id: __init__.py,v 1.11 2012-03-23 14:06:31 ivan Exp $
+$Date: 2012-03-23 14:06:31 $
 """
 __version__ = "3.0"
 
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/initialcontext.py pyRdfa/initialcontext.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/initialcontext.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/initialcontext.py	2012-04-03 00:26:42.781131661 +0100
@@ -13,8 +13,8 @@
 """
 
 """
-$Id: initialcontext.py,v 1.7 2012/03/16 16:46:41 ivan Exp $
-$Date: 2012/03/16 16:46:41 $
+$Id: initialcontext.py,v 1.7 2012-03-16 16:46:41 ivan Exp $
+$Date: 2012-03-16 16:46:41 $
 """
 
 class Wrapper :
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/options.py pyRdfa/options.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/options.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/options.py	2012-04-03 00:26:42.781131661 +0100
@@ -11,7 +11,7 @@
 """
 
 """
-$Id: options.py,v 1.12 2012/03/23 14:06:25 ivan Exp $ $Date: 2012/03/23 14:06:25 $
+$Id: options.py,v 1.12 2012-03-23 14:06:25 ivan Exp $ $Date: 2012-03-23 14:06:25 $
 """
 
 import sys, datetime
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/parse.py pyRdfa/parse.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/parse.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/parse.py	2012-04-03 00:26:42.781131661 +0100
@@ -15,8 +15,8 @@
 """
 
 """
-$Id: parse.py,v 1.13 2012/03/23 14:06:25 ivan Exp $
-$Date: 2012/03/23 14:06:25 $
+$Id: parse.py,v 1.13 2012-03-23 14:06:25 ivan Exp $
+$Date: 2012-03-23 14:06:25 $
 """
 
 import sys
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/property.py pyRdfa/property.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/property.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/property.py	2012-04-03 00:26:42.781131661 +0100
@@ -15,8 +15,8 @@
 """
 
 """
-$Id: property.py,v 1.9 2012/03/23 14:06:25 ivan Exp $
-$Date: 2012/03/23 14:06:25 $
+$Id: property.py,v 1.9 2012-03-23 14:06:25 ivan Exp $
+$Date: 2012-03-23 14:06:25 $
 """
 
 import re
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/py3compat.py pyRdfa/py3compat.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/py3compat.py	1970-01-01 01:00:00.000000000 +0100
+++ pyRdfa/py3compat.py	2012-04-03 00:26:42.777131637 +0100
@@ -0,0 +1,102 @@
+"""
+Utility functions and objects to ease Python 3 compatibility.
+"""
+import sys
+
+try:
+    from functools import wraps
+except ImportError:
+    # No-op wraps decorator
+    def wraps(f):
+        def dec(newf): return newf
+        return dec
+
+def cast_bytes(s, enc='utf-8'):
+    if isinstance(s, unicode):
+        return s.encode(enc)
+    return s
+
+PY3 = (sys.version_info[0] >= 3)
+
+def _modify_str_or_docstring(str_change_func):
+    @wraps(str_change_func)
+    def wrapper(func_or_str):
+        if isinstance(func_or_str, str):
+            func = None
+            doc = func_or_str
+        else:
+            func = func_or_str
+            doc = func.__doc__
+        
+        doc = str_change_func(doc)
+        
+        if func:
+            func.__doc__ = doc
+            return func
+        return doc
+    return wrapper
+    
+if PY3:
+    # Python 3:
+    # ---------
+    def b(s):
+        return s.encode('ascii')
+    
+    bytestype = bytes
+    
+    # Abstract u'abc' syntax:
+    @_modify_str_or_docstring
+    def format_doctest_out(s):
+        """Python 2 version
+        "%(u)s'abc'" --> "'abc'"
+        "%(b)s'abc'" --> "b'abc'"
+        "55%(L)s"    --> "55"
+        
+        Accepts a string or a function, so it can be used as a decorator."""
+        return s % {'u':'', 'b':'b', 'L':''}
+    
+    def type_cmp(a, b):
+        """Python 2 style comparison based on type"""
+        ta, tb = type(a).__name__, type(b).__name__
+        # Ugly hack: some tests rely on tuple sorting before unicode, and I
+        # don't know if that's important. Better retain it for now.
+        if ta == 'str':
+            ta = 'unicode'
+        if tb == 'str':
+            tb = 'unicode'
+        # return 1 if ta > tb else -1 if ta < tb else 0
+        if ta > tb:
+            return 1
+        elif ta < tb:
+            return -1
+        else:
+            return 0
+
+else:
+    # Python 2
+    # --------
+    def b(s):
+        return s
+    
+    bytestype = str
+    
+    # Abstract u'abc' syntax:
+    @_modify_str_or_docstring
+    def format_doctest_out(s):
+        """Python 2 version
+        "%(u)s'abc'" --> "u'abc'"
+        "%(b)s'abc'" --> "'abc'"
+        "55%(L)s"    --> "55L"
+        
+        Accepts a string or a function, so it can be used as a decorator."""
+        return s % {'u':'u', 'b':'', 'L':'L'}
+    
+    def type_cmp(a, b):
+        # return 1 if a > b else -1 if a < b else 0
+        if a > b:
+            return 1
+        elif a < b:
+            return -1
+        else:
+            return 0
+
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/rdfs/__init__.py pyRdfa/rdfs/__init__.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/rdfs/__init__.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/rdfs/__init__.py	2012-04-03 00:26:42.777131637 +0100
@@ -11,7 +11,7 @@
 """
 
 """
-$Id: __init__.py,v 1.3 2012/03/23 14:06:38 ivan Exp $ $Date: 2012/03/23 14:06:38 $
+$Id: __init__.py,v 1.3 2012-03-23 14:06:38 ivan Exp $ $Date: 2012-03-23 14:06:38 $
 
 """
 
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/rdfs/process.py pyRdfa/rdfs/process.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/rdfs/process.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/rdfs/process.py	2012-04-03 00:26:42.777131637 +0100
@@ -8,7 +8,7 @@
 """
 
 """
-$Id: process.py,v 1.7 2012/03/23 14:06:38 ivan Exp $ $Date: 2012/03/23 14:06:38 $
+$Id: process.py,v 1.7 2012-03-23 14:06:38 ivan Exp $ $Date: 2012-03-23 14:06:38 $
 
 """
 
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/serializers/prettyXMLserializer_3.py pyRdfa/serializers/prettyXMLserializer_3.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/serializers/prettyXMLserializer_3.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/serializers/prettyXMLserializer_3.py	2012-04-03 13:21:22.999614600 +0100
@@ -173,7 +173,7 @@
             if bnode not in self.__serialized:
                 self.subject(subject, 1)
         writer.pop(RDF.RDF)
-        stream.write("\n")
+        stream.write("\n".encode('utf-8'))
 
         # Set to None so that the memory can get garbage collected.
         self.__serialized = None
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/serializers/prettyXMLserializer.py pyRdfa/serializers/prettyXMLserializer.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/serializers/prettyXMLserializer.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/serializers/prettyXMLserializer.py	2012-04-03 13:02:06.001877368 +0100
@@ -9,9 +9,9 @@
 
 from rdflib	   import URIRef, Literal, BNode
 from rdflib.util  import first, uniq, more_than
-from rdflib.Graph import Seq
+from rdflib.graph import Seq
 
-from rdflib.syntax.serializers	import Serializer
+from rdflib.serializer import Serializer
 from pyRdfa.serializers.XMLWriter import XMLWriter
 
 XMLLANG = "http://www.w3.org/XML/1998/namespacelang"
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/serializers/turtleserializer.py pyRdfa/serializers/turtleserializer.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/serializers/turtleserializer.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/serializers/turtleserializer.py	2012-04-03 14:07:47.181420623 +0100
@@ -8,12 +8,12 @@
 import urlparse
 from xml.sax.saxutils 			import escape, quoteattr
 
-from rdflib.BNode 				import BNode
-from rdflib.Literal 			import Literal
-from rdflib.URIRef 				import URIRef
-from rdflib.syntax.xml_names 	import split_uri 
+from rdflib import BNode
+from rdflib import Literal
+from rdflib import URIRef
+from rdflib.namespace import split_uri 
 
-from rdflib.syntax.serializers.RecursiveSerializer import RecursiveSerializer
+from rdflib.plugins.serializers.turtle import RecursiveSerializer
 from rdflib.exceptions import Error
 
 from rdflib import RDF, RDFS
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/state.py pyRdfa/state.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/state.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/state.py	2012-04-03 00:26:42.781131661 +0100
@@ -17,8 +17,8 @@
 """
 
 """
-$Id: state.py,v 1.13 2012/03/23 14:06:25 ivan Exp $
-$Date: 2012/03/23 14:06:25 $
+$Id: state.py,v 1.13 2012-03-23 14:06:25 ivan Exp $
+$Date: 2012-03-23 14:06:25 $
 """
 
 import rdflib
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/termorcurie.py pyRdfa/termorcurie.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/termorcurie.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/termorcurie.py	2012-04-03 00:26:42.781131661 +0100
@@ -19,8 +19,8 @@
 """
 
 """
-$Id: termorcurie.py,v 1.7 2012/03/23 14:06:25 ivan Exp $
-$Date: 2012/03/23 14:06:25 $
+$Id: termorcurie.py,v 1.7 2012-03-23 14:06:25 ivan Exp $
+$Date: 2012-03-23 14:06:25 $
 """
 
 import re, sys
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/DublinCore.py pyRdfa/transform/DublinCore.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/DublinCore.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/transform/DublinCore.py	2012-04-03 00:26:42.777131637 +0100
@@ -18,8 +18,8 @@
 """
 
 """
-@version: $Id: DublinCore.py,v 1.4 2012/01/18 14:16:44 ivan Exp $
-$Date: 2012/01/18 14:16:44 $
+@version: $Id: DublinCore.py,v 1.4 2012-01-18 14:16:44 ivan Exp $
+$Date: 2012-01-18 14:16:44 $
 """
 
 def DC_transform(html, options, state) :
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/lite.py pyRdfa/transform/lite.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/lite.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/transform/lite.py	2012-04-03 00:26:42.777131637 +0100
@@ -5,8 +5,8 @@
 @license: This software is available for use under the
 U{W3C® SOFTWARE NOTICE AND LICENSE<href="http://www.w3.org/Consortium/Legal/2002/copyright-software-20021231">}
 @contact: Ivan Herman, ivan@w3.org
-@version: $Id: lite.py,v 1.7 2012/02/24 09:25:44 ivan Exp $
-$Date: 2012/02/24 09:25:44 $
+@version: $Id: lite.py,v 1.7 2012-02-24 09:25:44 ivan Exp $
+$Date: 2012-02-24 09:25:44 $
 """
 
 non_lite_attributes = ["about","inlist","datatype","rev","rel"]
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/metaname.py pyRdfa/transform/metaname.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/metaname.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/transform/metaname.py	2012-04-03 00:26:42.777131637 +0100
@@ -7,8 +7,8 @@
 @license: This software is available for use under the
 U{W3C® SOFTWARE NOTICE AND LICENSE<href="http://www.w3.org/Consortium/Legal/2002/copyright-software-20021231">}
 @contact: Ivan Herman, ivan@w3.org
-@version: $Id: metaname.py,v 1.3 2012/01/18 14:16:45 ivan Exp $
-$Date: 2012/01/18 14:16:45 $
+@version: $Id: metaname.py,v 1.3 2012-01-18 14:16:45 ivan Exp $
+$Date: 2012-01-18 14:16:45 $
 """
 
 def meta_transform(html, options, state) :
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/OpenID.py pyRdfa/transform/OpenID.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/OpenID.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/transform/OpenID.py	2012-04-03 00:26:42.777131637 +0100
@@ -14,8 +14,8 @@
 """
 
 """
-$Id: OpenID.py,v 1.4 2012/01/18 14:16:44 ivan Exp $
-$Date: 2012/01/18 14:16:44 $
+$Id: OpenID.py,v 1.4 2012-01-18 14:16:44 ivan Exp $
+$Date: 2012-01-18 14:16:44 $
 """
 
 OPENID_NS = "http://xmlns.openid.net/auth#"
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/toplevelabout.py pyRdfa/transform/toplevelabout.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/transform/toplevelabout.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/transform/toplevelabout.py	2012-04-03 00:26:42.777131637 +0100
@@ -13,8 +13,8 @@
 """
 
 """
-$Id: toplevelabout.py,v 1.4 2012/03/08 10:54:59 ivan Exp $
-$Date: 2012/03/08 10:54:59 $
+$Id: toplevelabout.py,v 1.4 2012-03-08 10:54:59 ivan Exp $
+$Date: 2012-03-08 10:54:59 $
 """
 
 def top_about(root, options, state) :
diff -uNr ../aa-shed/pyRdfa-3.2.0/pyRdfa/utils.py pyRdfa/utils.py
--- ../aa-shed/pyRdfa-3.2.0/pyRdfa/utils.py	2012-03-23 14:32:44.000000000 +0000
+++ pyRdfa/utils.py	2012-04-03 00:26:42.777131637 +0100
@@ -13,8 +13,8 @@
 """
 
 """
-$Id: utils.py,v 1.7 2012/03/23 14:06:25 ivan Exp $
-$Date: 2012/03/23 14:06:25 $
+$Id: utils.py,v 1.7 2012-03-23 14:06:25 ivan Exp $
+$Date: 2012-03-23 14:06:25 $
 """
 import os, os.path, sys, imp, datetime
 import urllib, urlparse, urllib2
